modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Actions.tsx

```diff
@@ -66,21 +73,266 @@ const TYPES = [
+			);
+		}
+
+		setFDSActions([...notOrdered, ...ordered]);
+
+		setLoading(false);
+	};
+
+	const saveFDSAction = async () => {
+		setSaveButtonDisabled(true);
+
+		const body = {
+			[OBJECT_RELATIONSHIP.FDS_VIEW_FDS_ACTION_ID]: fdsView.id,
+			icon: iconSymbol,
+			type,
+			url,
+		} as any;
+
+		if (Liferay.FeatureFlags['LPS-172017']) {
+			body.confirmationMessage_i18n = confirmationMessageTranslations;
+			body.label_i18n = labelTranslations;
+		}
+		else {
+			body.confirmationMessage = confirmationMessage;
+			body.label = labelTranslations;
+		}
+
+		const response = await fetch(API_URL.FDS_ACTIONS, {
+			body: JSON.stringify(body),
+			headers: {
+				'Accept': 'application/json',
+				'Content-Type': 'application/json',
+			},
+			method: 'POST',
+		});
+
+		if (!response.ok) {
+			setSaveButtonDisabled(false);
+
+			openDefaultFailureToast();
+
+			return;
+		}
+
+		await response.json();
+
+		setSaveButtonDisabled(false);
+
+		openDefaultSuccessToast();
+
+		setActiveSection(SECTIONS.ACTIONS);
+
+		getFDSActions();
+	};
+
+	const updateFDSActionsOrder = async () => {
+		const response = await fetch(
+			`${API_URL.FDS_VIEWS}/by-external-reference-code/${fdsView.externalReferenceCode}`,
+			{
+				body: JSON.stringify({
+					fdsActionsOrder: newActionsOrder,
+				}),
+				headers: {
+					'Accept': 'application/json',
+					'Content-Type': 'application/json',
+				},
+				method: 'PATCH',
+			}
+		);
+
+		if (!response.ok) {
+			openDefaultFailureToast();
+
+			return;
+		}
+
+		const responseJSON = await response.json();
+
+		const fdsFiltersOrder = responseJSON?.fdsActionsOrder;
+
+		if (fdsFiltersOrder && fdsFiltersOrder === newActionsOrder) {
+			openDefaultSuccessToast();
+
+			setNewActionsOrder('');
+		}
+		else {
+			openDefaultFailureToast();
+		}
+	};
+
+	useEffect(() => {
+		getFDSActions();
+
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, []);
+
+	useEffect(() => {
+		const getIcons = async () => {
+			const response = await fetch(spritemap);
+
+			const responseText = await response.text();
+
+			if (responseText.length) {
+				const spritemapDocument = new DOMParser().parseFromString(
+					responseText,
+					'text/xml'
+				);
+
+				const symbolElements = spritemapDocument.querySelectorAll(
+					'symbol'
+				);
+
+				const iconSymbols = Array.from(symbolElements!).map(
+					(element) => ({
+						label: element.id,
+						value: element.id,
+					})
+				);
+
+				setAvailableIconSymbols(iconSymbols);
+			}
+		};
+
+		getIcons();
+	}, [spritemap]);
+
+	const labelFormElementId = `${namespace}Label`;
+	const iconFormElementId = `${namespace}Icon`;
+	const typeFormElementId = `${namespace}Type`;
+	const urlFormElementId = `${namespace}URL`;
+	const confirmationMessageFormElementId = `${namespace}ConfirmationMessage`;
+	const confirmationMessageTypeFormElementId = `${namespace}ConfirmationMessageType`;
+
+	const ConfirmationMessageRow = () => (
```

Comment:
Buying space. No state in this component, so we should be safe,

Reply:
I spent quite some time debugging this, but abstracting the row into this component bugged the `InputLocalized` component inside it, so I will be removing this abstraction in a follow-up. cc @dsanz this is the thing we talked about on friday